bitcoin/src/blockdata/witness.rs

```diff
@@ -246,29 +278,69 @@ impl Witness {
         if self.witness_elements <= 1 {
             None
         } else {
-            self.element_at(self.second_to_last)
+            self.nth(self.witness_elements - 2)
         }
     }
+
+    /// Return the nth element in the witness, if any
+    pub fn nth(&self, index: usize) -> Option<&[u8]> {
+        let pos = decode_cursor(&self.content, self.indices_start, index)?;
+        self.element_at(pos)
+    }
+
+    /// Get Tapscript following BIP341 rules regarding accounting for an annex.
+    /// This does not guarantee that this represents a P2TR [`Witness`].
+    /// It merely gets the second to last or third to last element depending
+    /// on the first byte of the last element being equal to 0x50.
```

Comment:
Perhaps
```
    /// Get Tapscript following BIP341 rules regarding accounting for an annex.
    ///
    /// Does not guarantee that this [`Witness`] is P2TR. Merely gets the second to last or third to
    /// last element depending on the first byte of the last element being equal to 0x50.
```

Mainly adding the newline between brief description and long description, other changes are merely suggestions.

Reply:
Thanks!